woocommerce/woocommerce
# Add new import filter in addition to existing action

### All Submissions:






### Changes proposed in this Pull Request:

Currently there is no way to change the data fed into the importer. An example use of would is the following filter to prevent inserting empty strings if the name in some column is missing. As this shouldn't be default behaviour but (in my case for example) can be the desired functionality, a filter would be nice to have.

    add_filter('woocommerce_product_import_before_process_item', "product_import_before_process_item", 10, 1);
    function product_import_before_process_item($data){
        //modify the data before importing e.g. prevent inserting an empty name
        foreach($data as $key => $value){
            if(isset($data[$key]) && empty($value)){
                unset($data[$key]);
            }
        }
        return $data;
    }

### How to test the changes in this Pull Request:

0. It's just one line of code, the easiest way would just be to check if I made any typos / if the filter name should be different. As this pull request format is given, I'll write some test instructions anyway:
1. Insert example code that is provided above.
2. Create a csv with two products, one with the name column being set, the other with an empty string in it
3. Import the file using the woocommerce csv importer

### Other information:




### Changelog entry

> Added "woocommerce_product_import_before_process_item" filter in addition to the existing action
